From 740f1c7ff7d0a429e79d5bf075c162314ca6fd66 Mon Sep 17 00:00:00 2001
From: Adithya R <gh0strider.2k18.reborn@gmail.com>
Date: Thu, 19 Jan 2023 14:37:43 +0530
Subject: [PATCH 30/34] SystemUI: Remove visibility check in setting QSCarrier
 color

This fixes a corner case where the signal icon color is incorrect:
Have dual sim -> switch to dark theme -> turn off one sim -> switch
to light theme -> turn the other sim back on. QS carrier signal
is colored white instead of black.

Ensure that the signal icon color is always up to date, by removing
visibility check, to avoid this issue.

Fixes: 816b8ddf ("Follow light/dark theme in SplitShade Header")
Change-Id: I092c06053fc4bc8d9ca51d1d31128da27ef6a823
---
 .../com/android/systemui/shade/carrier/ShadeCarrier.java   | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/packages/SystemUI/src/com/android/systemui/shade/carrier/ShadeCarrier.java b/packages/SystemUI/src/com/android/systemui/shade/carrier/ShadeCarrier.java
index 24a1e7cda1fd..727a187d20f7 100644
--- a/packages/SystemUI/src/com/android/systemui/shade/carrier/ShadeCarrier.java
+++ b/packages/SystemUI/src/com/android/systemui/shade/carrier/ShadeCarrier.java
@@ -153,11 +153,8 @@ public class ShadeCarrier extends LinearLayout {
     }
 
     public void updateColors(ColorStateList colorStateList) {
-        final boolean visible = !mIsSingleCarrier;
-        if (visible) {
-            mMobileRoaming.setImageTintList(colorStateList);
-            mMobileSignal.setImageTintList(colorStateList);
-        }
+        mMobileRoaming.setImageTintList(colorStateList);
+        mMobileSignal.setImageTintList(colorStateList);
     }
 
     @VisibleForTesting
-- 
2.34.1

